tests/src/main/java/alluxio/master/backcompat/BackwardsCompatibilityJournalGenerator.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.backcompat;
+
+import alluxio.AlluxioURI;
+import alluxio.Constants;
+import alluxio.ProjectConstants;
+import alluxio.master.backcompat.ops.AsyncPersist;
+import alluxio.master.backcompat.ops.CreateDirectory;
+import alluxio.master.backcompat.ops.CreateFile;
+import alluxio.master.backcompat.ops.DeleteFile;
+import alluxio.master.backcompat.ops.Mount;
+import alluxio.master.backcompat.ops.PersistDirectory;
+import alluxio.master.backcompat.ops.PersistFile;
+import alluxio.master.backcompat.ops.RenameFile;
+import alluxio.master.backcompat.ops.SetAcl;
+import alluxio.multi.process.MultiProcessCluster;
+import alluxio.multi.process.PortCoordination;
+import alluxio.security.LoginUser;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.google.common.collect.ImmutableList;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.util.List;
+
+/**
+ * Generates journals for consumption by the BackwardsCompatibilityIntegrationTest.
+ *
+ * This class
+ * 1. starts a new cluster
+ * 2. runs all the test operations listed in OPS
+ * 3. takes a journal backup at stores it to src/test/resources/old_journals
+ * 4. stops the cluster
+ * 5. copies the journal to src/test/resources/old_journals
+ *
+ * Later, BackwardsCompatibilityIntegrationTest will start clusters from the backups and copies
+ * in old_journals, and validate that the operations' checks pass.
+ */
+public final class BackwardsCompatibilityJournalGenerator {
+  // Path relative to tests/src/test
+  public static final String OLD_JOURNALS_RESOURCE = "src/test/resources/old_journals";
+
+  public static final List<TestOp> OPS = ImmutableList.<TestOp>builder()
+      .add(new CreateDirectory(),
+          new CreateFile(),
+          new Mount(),
+          new AsyncPersist(),
+          new DeleteFile(),
+          new PersistFile(),
+          new PersistDirectory(),
+          new RenameFile(),
+          new SetAcl()
+      ).build();
```

Comment:
There are some operations which may journal that are not listed here (ie. getStatus)? Alternatively, should we just create an old journal for each version and commit it to the repo?

Reply:
there aren't `getStatus` journal entries - `getStatus` just triggers the creation of other types of entries already covered here. I think it would be nice to add an operation for `getStatus` in the future for better coverage, but I don't see it as essential

Reply:
Okay, in that case would it be better to directly invoke the methods writing journal entries, as opposed to using the `FileSystem` client?

Reply:
I'd prefer to stick with the public API
- At the end of the day it's the part that is supposed to be backwards compatible
- Public API will also result in more realistic entries - if we write entries directly we might be testing entries that could never actually be written
- Using internal APIs makes the test harder to maintain across versions since the APIs could change

Reply:
I think that's the better approach, and it sounds like you are on board with adding the other client operations after the initial commit.